test/plugin/example/example_plugin.go

```diff
@@ -2,54 +2,91 @@ package main
 // and is expected to return the target service connection.
 //
 // This example connector works as follows:
 // 1. Waits for the initial message from the client
-// 2. Connect to a target service whose address is the value of the secret identified by
-// the key "address"
-// 3. Inject credentials from a secret identified by the key "auth"
+// 2. Connect to a target service whose address is the value of the credential identified
+// by the key "address"
+// 3. Inject credentials from a credential identified by the key "auth"
 // 4. Write the initial message from the client with some modification
-func connectorFunc(clientConn net.Conn, secrets connector.SecretsByID) (net.Conn, error) {
+func (c *Connector) Connect(
+	clientConn net.Conn,
+	credentialValuesByID connector.CredentialValuesByID,
+) (net.Conn, error) {
+	c.logger.Debugln("waiting for initial write from client")
 	clientInitMsg, _, err := bufio.NewReader(clientConn).ReadLine()
 	if err != nil {
 		return nil, err
 	}
 
-	conn, err := net.Dial("tcp", string(secrets["address"]))
+
+	c.logger.Debugln("dialing target service")
+	conn, err := net.Dial("tcp", string(credentialValuesByID["address"]))
 	if err != nil {
 		return nil, err
 	}
 
-	_, err = conn.Write([]byte("credential injection: " + string(secrets["auth"]) + "\n"))
+	c.logger.Debugln("sending packet with injected credentials to target service")
+	credInjectionPacket := []byte(
+		fmt.Sprintf(
+			"credential injection: %s\n",
+			string(credentialValuesByID["auth"]),
+		),
+	)
+	_, err = conn.Write(credInjectionPacket)
 	if err != nil {
 		return nil, err
 	}
-	_, err = conn.Write([]byte("initial message from client: " + string(clientInitMsg) + "\n"))
+
+	c.logger.Debugln("sending modified client initial packet to target service")
+	initMsgPacket := []byte(
+		fmt.Sprintf(
+			"initial message from client: %s\n",
+			string(clientInitMsg),
+		),
+	)
+	_, err = conn.Write(initMsgPacket)
 	if err != nil {
 		return nil, err
 	}
 
+	c.logger.Debugln("successfully connected to target service")
 	return conn, nil
 }
 
-// NewConnector is required method on the tcp.Plugin interface. It returns a
+// NewConnector is a required method on the tcp.Plugin interface. It returns a
 // tcp.Connector.
 //
 // The single argument passed in is of type connector.Resources. It contains
-// connector-specific config and a logger. This particular plugin ignores these.
-func (examplePlugin *examplePlugin) NewConnector(_ connector.Resources) tcp.Connector {
-	return connectorFunc
+// connector-specific config and a logger.
+func NewConnector(conRes connector.Resources) tcp.Connector {
+	return (&Connector{
+		logger:   conRes.Logger(),
+	}).Connect
+}
+
+// pluginWrapper is a wrapper type that makes it possible for the NewConnector func
+// to stand alone as a tcp.Plugin
+type pluginWrapper func (connector.Resources) tcp.Connector
+func (pw pluginWrapper) NewConnector(cr connector.Resources) tcp.Connector {
+	return pw(cr)
```

Comment:
I think the name is `pluginWrapper` is inaccurate, since we're not actually wrapping anything here.  It's simply a type definition.  The type represents a "connector contstructor".  A couple options that would be accurate imo:
- `type NewConnectorFunc func (connector.Resources) tcp.Connector`... following the rule of only using the `Func` suffix when defining a function _type_.  So here's we'd be saying "Any function of the _type_ `NewConnector` has a method `NewConnector` that simply calls itself".  A bit meta but it makes sense.
- `type ConnectorConstructor func (connector.Resources) tcp.Connector`... reads a bit more naturally but perhaps less conventional. And here we'd be saying "Any function of the _type_ `ConnectorConstructor` has a... well, constructor method called `NewConnector` (per Go convention for constructor naming) that simply calls itself".

Reply:
Added `ConnectorConstructor` type to public `plugin/connector` packge